File: TowerOfHanoi/solve_tower_of_hanoi.py
def solve_tower_of_hanoi(n, source, destination, auxiliary):
    if n == 1:
        print(f"Move disk 1 from {source} to {destination}")
        return 1
    else:
        moves = 0
        moves += solve_tower_of_hanoi(n - 1, source, auxiliary, destination)
        print(f"Move disk {n} from {source} to {destination}")
        moves += 1
        moves += solve_tower_of_hanoi(n - 1, auxiliary, destination, source)
        return moves

if __name__ == "__main__":
    number_of_disks = int(input("Please enter the number of disks: "))
    total_moves = solve_tower_of_hanoi(number_of_disks, 'A', 'C', 'B')
    print(f"Total number of moves: {total_moves}")

File: TowerOfHanoi/solveTowerOfHanoi.c
#include <stdio.h>

int solveTowerOfHanoi(int n, char source, char destination, char auxiliary) {
    if (n == 1) {
        printf("Move disk 1 from %c to %c\n", source, destination);
        return 1;
    } else {
        int moves = 0;
        moves += solveTowerOfHanoi(n - 1, source, auxiliary, destination);
        printf("Move disk %d from %c to %c\n", n, source, destination);
        moves++;
        moves += solveTowerOfHanoi(n - 1, auxiliary, destination, source);
        return moves;
    }
}

int main() {
    int numberOfDisks;
    printf("Please enter the number of disks: ");
    scanf("%d", &numberOfDisks);
    
    int totalMoves = solveTowerOfHanoi(numberOfDisks, 'A', 'C', 'B');
    printf("Total number of moves: %d\n", totalMoves);
    
    return 0;
}
